internal/dataplane/kongstate/kongstate.go

```diff
@@ -690,3 +683,301 @@ func maybeLogKongIngressDeprecationError(logger logr.Logger, services []*corev1.
 		}
 	}
 }
+
+// FillCustomEntities fills custom entities in KongState.
+func (ks *KongState) FillCustomEntities(
+	logger logr.Logger,
+	s store.Storer,
+	failuresCollector *failures.ResourceFailuresCollector,
+	schemaGetter SchemaGetter,
+	workspace string,
+) {
+	logger = logger.WithName("fillCustomEntities")
+	entities := s.ListKongCustomEntities()
+	if len(entities) == 0 {
+		return
+	}
```

Comment:
No need for logger when `len(entities) == 0`

```suggestion
	entities := s.ListKongCustomEntities()
	if len(entities) == 0 {
		return
	}
	logger = logger.WithName("fillCustomEntities")
```

Reply:
Done.